src/mscorlib/src/System/Decimal.cs

```diff
@@ -338,14 +336,18 @@ void IDeserializationCallback.OnDeserialization(Object sender)
         // Constructs a Decimal from its constituent parts.
         private Decimal(int lo, int mid, int hi, int flags)
         {
-            if ((flags & ~(SignMask | ScaleMask)) == 0 && (flags & ScaleMask) <= (28 << 16))
-            {
-                this.lo = lo;
-                this.mid = mid;
-                this.hi = hi;
-                this.flags = flags;
-                return;
-            }
+            if (!((flags & ~(SignMask | ScaleMask)) == 0 && (flags & ScaleMask) <= (28 << 16)))
```

Comment:
Is this intentional?

Reply:
It was; but is also completely unnecessary as the function doesn't return anything - will revert